ipn/ipnlocal/local.go

```diff
@@ -1742,6 +1742,24 @@ func (b *LocalBackend) readPoller() {
 func (b *LocalBackend) WatchNotifications(ctx context.Context, mask ipn.NotifyWatchOpt, fn func(roNotify *ipn.Notify) (keepGoing bool)) {
 	ch := make(chan *ipn.Notify, 128)
 
+	origFn := fn
+	if mask&ipn.NotifyNoPrivateKeys != 0 {
+		fn = func(n *ipn.Notify) bool {
+			if n.NetMap == nil || n.NetMap.PrivateKey.IsZero() {
+				return origFn(n)
+			}
+
+			// The netmap in n is shared across all watchers, so to mutate it for a
+			// single watcher we have to clone the notify and the netmap. We can
+			// make shallow clones, at least.
+			nm := *n.NetMap
+			r := *n
+			r.NetMap = &nm
+			r.NetMap.PrivateKey = key.NodePrivate{}
```

Comment:
Maybe variable names `nm2` (netmap 2) and `n2` (notify 2) ?

Reply:
Done. And one more round through the dog-slow CI, yay!